### PR TITLE
Add testing spec skeleton

### PR DESCRIPTION
**Pull request by UlisesGascon**
### Main changes

- Added basic support for Node.js App (testing, linting, nvm...)
- Added basic pipeline to review PRs
- Added sample spec for documentation and tests using JSON schema and Jest
- Added basic documentation

### Context

This skeleton is based on [POC-semver-and-json-schemas](https://github.com/UlisesGascon/POC-semver-and-json-schemas)

### Changelog

- 96add03 chore: added basic Node.js app skeleton by @UlisesGascon
- 128445b chore: added basic dependencies together with linting and testing setup by @UlisesGascon
- eeb1785 feat: added basic json schema by @UlisesGascon
- 597d5fb feat: added basic spec by @UlisesGascon
- e51dad2 feat: added basic spec tests by @UlisesGascon
- 2e1e043 docs: updated documentation by @UlisesGascon
- bc4eb01 feat: added basic pipeline by @UlisesGascon

### CODE REVIEW

File: src/schemas/pkg.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
+    "version": "1.0.0",
+    "name": "package.json specifying",
+    "schema": {
+      "title": "package.json specifying",
+      "type": "object",
+      "properties": {
+        "name": {
+          "description": "The name is what your thing is called",
```
Comment (styfle) [inline src/schemas/pkg.json]:
```suggestion
          "description": "The name of your package",
```

Comment (UlisesGascon) [reply src/schemas/pkg.json]: thanks @styfle

File: src/spec.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+## The specification
+
+
+### Name
+
+The `name` is what your thing is called.
+
+Details:
+- The `name` is an optional property.
+- The `name` must be less than or equal to 214 characters.
```
Comment (styfle) [inline src/spec.md:13]: Perhaps the existing npm docs can be copied as a starting point.

https://docs.npmjs.com/cli/v9/configuring-npm/package-json

Comment (UlisesGascon) [reply src/spec.md:13]: yes, I was using part of it as a reference. I can port the docs easily if we agreed on it at https://github.com/wintercg/proposal-package-json/issues/1#issuecomment-1540929817 in a separate PR 😄

**Comment by wesleytodd** [comment]
I think this PR should not land as is, and I honestly think the presence of this already here is jumping the gun. @UlisesGascon I am super happy to see your excitement, but I am worried that jumping right into implementation might cause legal or process issues for the effort. I think a better way to approach this would be to propose a set of things you would like to PR in an issue and then as this effort gets off the ground we can discuss and make sure we are not introducing any IP restrictions which would block this getting actual standards bodies involved.

**Comment by UlisesGascon** [comment]
Thanks for the advice @wesleytodd 😉. I will close the PR for now until we are clear on how to move forward

**Pull request closed without merging**